lib/wallaroo/core/step/step_phase.pony

```diff
@@ -311,8 +311,15 @@ class _RecoveringStepPhase is StepPhase
   fun ref receive_barrier(step_id: RoutingId, producer: Producer,
     barrier_token: BarrierToken)
   =>
-    @printf[I32]("Ignoring non-rollback barrier in _RecoveringStepPhase: %s\n"
-      .cstring(), barrier_token.string().cstring())
+    match barrier_token
```

Comment:
I think this doesn't respect the priority ordering we need to enforce.  My parallel fix accounts for that.

Reply:
It's there to permit a checkpoint operation in between the autoscale and autoscale-resume ops.  If there isn't a mechanism to permit a `CheckpointBarrierToken` through, then it will be queued until after autoscale-resume is done, and that's definitely not when we want to perform that checkpoint.